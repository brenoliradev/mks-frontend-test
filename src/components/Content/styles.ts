import styled from 'styled-components'

export const ContentWrapper = styled.div`
  grid-area: main;
  height: calc(100vh - 135px);
  width: 90%;
  max-width: 938px;
  display: grid;
  grid-cols: 4;
  justify-content: center;
  align-items: center;
  margin: auto;
`
