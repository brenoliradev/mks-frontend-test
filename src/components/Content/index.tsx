import { ContentWrapper } from './styles'
import { useFetch } from '../../hooks'
import { useEffect, useMemo } from 'react'
import { ProductProps } from '../../typings/api'

type ProductReturn = { count: number; products: ProductProps[] }

const Content = () => {
  const { callFetch, isLoading, myRes } = useFetch<ProductReturn>({
    url: 'https://mks-frontend-challenge-api.herokuapp.com/api/v1/products?page=1&rows=10&sortBy=id&orderBy=DESC',
  })

  // eslint-disable-next-line react-hooks/exhaustive-deps
  useEffect(() => callFetch(), [])

  const mapResult = useMemo(
    () =>
      myRes &&
      myRes?.products?.map(({ brand, description, id, name, price }) => (
        <div key={id}>{name}</div>
      )),
    [myRes],
  )

  return (
    <ContentWrapper>
      <h1>{isLoading && myRes ? 'Loading' : mapResult}</h1>
    </ContentWrapper>
  )
}

export default Content
