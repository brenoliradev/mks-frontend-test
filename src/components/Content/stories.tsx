import { Story, Meta } from '@storybook/react'

import Content from '.'

export default {
  component: Content,
  title: 'Content',
} as Meta

export const Basic: Story = () => <Content />
