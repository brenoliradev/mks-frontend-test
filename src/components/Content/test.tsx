import { render } from '@testing-library/react'

import Content from '.'

describe('<Content />', () => {
  it('should render the Content text', () => {
    render(<Content />)
  })

  it('should render container correctly', () => {
    const { container } = render(<Content />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Content />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#e5e5e5' })
  })
})
