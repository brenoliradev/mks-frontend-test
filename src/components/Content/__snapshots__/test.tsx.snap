// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> should render container correctly 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  min-height: calc(100vh - 135px);
  background: #e5e5e5;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  grid-area: main;
  width: 90%;
  max-width: 938px;
  display: grid;
  grid-template-columns: repeat(4,1fr);
  grid-template-rows: repeat(2,1fr);
  grid-column-gap: 23px;
  grid-row-gap: 31px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 3rem 0;
}

@media (max-width:1024px) {
  .c1 {
    grid-template-columns: repeat(3,1fr);
    grid-template-rows: repeat(3,1fr);
  }
}

@media (max-width:800px) {
  .c1 {
    grid-template-columns: repeat(2,1fr);
    grid-template-rows: repeat(4,1fr);
    grid-column-gap: 12px;
    grid-row-gap: 16px;
  }
}

<main
  class="c0"
>
  <div
    class="c1"
  />
</main>
`;
